source/parser.hera

```diff
@@ -2367,9 +2387,27 @@ MethodDefinition
   # NOTE: Not adding extra validation using PropertySetParameterList
   # NOTE: If this node layout changes, be sure to update `convertMethodTOFunction`
   MethodSignature:signature !PropertyAccess BracedOrEmptyBlock:block ->
+    let children = $0
+
+    if (hasAwait(block)) {
+      children = children.slice()
+      // get or set
+      if (signature.modifier) {
+        children.push({
+          type: "Error",
+          message: "Getters and setters cannot be async",
+        })
+      } else if(gatherRecursive(signature.modifier, (n) => n.type === "Async").length) {
+        // Do nothing, alread async
```

Comment:
```suggestion
        // Do nothing, already async
```

I wonder whether there should be an `async` attribute in the `signature` instead of having to search like this.